README.md

```diff
@@ -1 +1,72 @@
 # Goi
+
+## Deck Creation DSL
+### Example
+```
+Create Style mystyle:
+Color = Red
+Direction = Horizontal
+Align = Center
+
+Create Deck French with Tags language, hard, midterm using Style mystyle:  
+(1) Front:Back
+(2) Front:Back
+(3) Front:Back
+```
+
+### EBNF
+```
+PROGRAM ::= CREATE_STYLE* CREATE_DECK*
+
+CREATE_DECK ::= “Create Deck” NAME MODIFIERS”:” DECK
+MODIFIERS ::= (“with Tags” TAGS (“using Style” STYLE)?) | (“using Style” STYLE (“with Tags” TAGS)?)
+DECK ::= CARD+
+CARD ::= “(”NUMBER”) ” STRING “:” STRING
+TAGS ::= STRING”,”? TAGS
+
+CREATE_STYLE ::= “Create Style” NAME “:” STYLE
+STYLE ::= ATTRIBUTE “=” STRING
+ATTRIBUTE ::= “Direction” | “Font” | “Align” | “Color” | “Colour”
+
+NAME ::= STRING
+```
+
+## Interactive Prompt DSL
+### Examples
+```
+Show best scores for Decks: deck1, deck2 (Limit: 5)
+Show Sessions (Limit: 10)
+Show oldest Sessions (Limit: 10)
+Show best scores for Decks: deck1
+Show average time spent on Decks: deck2
+Start Session from Decks: deck1, deck2
+Start Session from Tags: hard, language
+Start Session from random cards from Decks: deck1, deck2
+
+```
+
+### EBNF
+```
+COMMANDS ::= (ACTION SELECTOR? SUBJECT LIMIT? | HELP | (LIST LIMIT?)
+ACTION ::= “Show” | “Start Session from”
+
+SELECTOR ::= (STAT “time spent on”) | 
+             (BEST_WORST “scores for”) | 
+             (BEST_WORST | “random” “cards from ”) | 
+             OLDEST_NEWEST
+STAT ::= “minimum” | “maximum” | “average” 
+OLDEST_NEWEST::= “oldest” | “newest”
+BEST_WORST ::= “best” | “worst”
```

Comment:
```suggestion
COMMANDS ::= ((SHOW | START_SESSION) SUBJECT LIMIT?) | HELP | LIST
SHOW ::= "Show" (STAT_TO_SHOW | SELECTED_CARDS)
STAT_TO_SHOW ::= STAT STAT_ITEM
STAT ::= "minimum" | "maximum" | "average"
STAT_ITEM ::= "time spent on" | "scores for"
SELECTED_CARDS::= CARD_FILTER "cards from"
CARD_FILTER ::= "best" |  "worst" | "random" | "oldest" | "newest"
START_SESSION ::= "Start Session from"  SELECTED_CARDS
```

Reply:
Mostly for refactoring/easier parsing and pre-empting commands like "Start session from minimum time spent on ... " and the like